### ISSUE TITLE
Support Managing Service Quotas

### ISSUE THREAD

**Issue by bflad** [terraform-providers/terraform-provider-aws#9118]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The new Service Quotas service provides a simplified method for requesting quota (limit) increases. Theoretically, Terraform can support this as a resource by submitting the request via the new API and reading back either request values and/or current service quotas (seems preferable!).

A separate feature request is created for managing Service Quotas Quota Request Templates and their associations for Organizations: #9119 

### New Resource(s)

* `aws_servicequotas_service_quota`
* `aws_servicequotas_service_quota_increase_request`

#### Option 1: Increase request managing resource

Simply submits quota increases. No lifecycle management for service quotas if they are increased in the future.

```hcl
# Relevant API Calls:
#   GetRequestedServiceQuotaChange
#   RequestServiceQuotaIncrease
resource "aws_servicequotas_service_quota_increase_request" "example" {
  desired_value = 123   # TypeFloat, Required, ForceNew
  quota_code    = "..." # TypeString, Required, ForceNew
  service_code  = "..." # TypeString, Required, ForceNew
}
```

#### Option 2: Service quota managing resource

Manages lifecycle of service quota. Requests quota increase if new value is greater than current value. Value is based on desired value of quota increase request if pending or case opened state, otherwise based on current service quota.

```hcl
# Relevant API Calls:
#   GetRequestedServiceQuotaChange
#   GetServiceQuota
#   RequestServiceQuotaIncrease
# Additional attributes:
#   request_id # TypeString, Computed
#   request_status # TypeString, Computed
resource "aws_servicequotas_service_quota" "example" {
  quota_code    = "..." # TypeString, Required, ForceNew
  service_code  = "..." # TypeString, Required, ForceNew
  value         = 123   # TypeFloat, Required
}
```

### References

* https://aws.amazon.com/about-aws/whats-new/2019/06/introducing-service-quotas-view-and-manage-quotas-for-aws-services-from-one-location/
* https://aws.amazon.com/blogs/mt/introducing-service-quotas-view-and-manage-your-quotas-for-aws-services-from-one-central-location/
* `GetRequestedServiceQuotaChange`
* `GetServiceQuota`
* `RequestServiceQuotaIncrease`

**Comment by bflad** [comment]
Draft implementation of option 2 submitted: #9121

**Comment by bflad** [comment]
The new `aws_servicequotas_service_quota` resource has been merged and will release with version 2.20.0 of the Terraform AWS Provider, later this week. 🎉
